p2p/conn/secret_connection_test.go

```diff
@@ -245,11 +245,11 @@ func TestSecretConnectionReadWrite(t *testing.T) {
 						}
 						*nodeReads = append(*nodeReads, string(readBuffer[:n]))
 					}
-					if err := nodeConn.PipeReader.Close(); err != nil {
-						t.Error(err)
-						return nil, err, true
-					}
-					return nil, nil, false
+					// if err := nodeConn.PipeReader.Close(); err != nil {
```

Comment:
remove?

Reply:
You can definitely remove without changing the existing semantics here. What the developer probably wanted, was that the connection gets closed on an error/when the control flow exits the loop. Sounds like this clsoing code should be moved to a defer (using the outer function though)?

Reply:
```go
defer func() {
				if err := nodeConn.PipeReader.Close(); err != nil {
					t.Error(err)
				}
			}()
```
on line 206?

Reply:
https://github.com/tendermint/tendermint/pull/3292/commits/b38cbae53df73ac9b712d56431968931baf14e7c